dev-tools/docker/linux_image/Dockerfile

```diff
@@ -82,3 +90,51 @@ RUN \
   cd .. && \
   rm -rf patchelf-0.10
 
+# Build Python 3.7
+# —-enable-optimizations for a stable/release build
+RUN \
+  cd ${build_dir} && \
+  wget --quiet -O - https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tgz | tar xzf - && \
+  cd Python-3.7.9/ && \  
+  ./configure --enable-optimizations && \ 
+  make altinstall && \
+  cd .. && rm -rf Python-3.7.9
+
+# Install Python dependencies 
+RUN \
+  /usr/local/bin/pip3.7 install numpy ninja pyyaml mkl mkl-include setuptools cmake cffi typing_extensions future six requests dataclasses  
+
+# Install CMake
+# v3.5 minimum is required
+ARG cmake_install_dir=/usr/local/cmake
```

Comment:
I am not sure this should be an `ARG`.  As things stand there doesn't seem to be any benefit in changing it, as we don't use CMake outside of this Dockerfile.  But if we _do_ start using it later on in our build process, if somebody changed it to a different value here then that would just create complication for our downstream build scripts.  The way our build environment is specified is that you _must_ install build tools into certain directories and then downstream scripts don't have to deal with uncertainty about where they are.

Reply:
I used an ARG to avoid copy and paste programming but yes I suppose it shouldn't be updatable by the outside world. 

ARG `build_dir` is ok as that directory is wiped in the final command anyway